src/diart/models.py

```diff
@@ -143,33 +150,27 @@ def from_pyannote(
         assert _has_pyannote, "No pyannote.audio installation found"
         return PyannoteEmbeddingModel(model, use_hf_token)
 
-    @abstractmethod
-    def forward(
-        self, waveform: torch.Tensor, weights: Optional[torch.Tensor] = None
-    ) -> torch.Tensor:
-        """
-        Forward pass of an embedding model with optional weights.
-
-        Parameters
-        ----------
-        waveform: torch.Tensor, shape (batch, channels, samples)
-        weights: Optional[torch.Tensor], shape (batch, frames)
-            Temporal weights for each sample in the batch. Defaults to no weights.
-
-        Returns
-        -------
-        speaker_embeddings: torch.Tensor, shape (batch, embedding_dim)
-        """
-        pass
-
 
 class PyannoteEmbeddingModel(EmbeddingModel):
     def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__(PyannoteLoader(model_info, hf_token))
 
-    def forward(
-        self,
-        waveform: torch.Tensor,
-        weights: Optional[torch.Tensor] = None,
+    def __call__(
+        self, waveform: torch.Tensor, weights: Optional[torch.Tensor] = None
     ) -> torch.Tensor:
-        return self.model(waveform, weights=weights)
+        # Normalize weights
+        if weights is not None:
+            weights -= weights.min(dim=1, keepdim=True).values
+            weights /= weights.max(dim=1, keepdim=True).values
+            weights.nan_to_num_(0.0)
+
+        if isinstance(self.model, nn.Module):
+            return super().__call__(waveform, weights)
+
+        else:
+            if weights is not None:
+                # Move to cpu for numpy conversion
+                weights = weights.to("cpu")
+            # Move to cpu for numpy conversion
+            waveform = waveform.to("cpu")
+            return torch.from_numpy(super().__call__(waveform, weights))
```

Comment:
This should all be simplified to:

```python
return super().__call__(waveform, weights)
```

as soon as we have a new pyannote release with the fix

Reply:
Okay, so we stall this until pyannote/pyannote-audio#1518 is merged?

Reply:
I would simplify the code here anyway and merge this PR once pyannote's is merged too. This is clearly an issue on pyannote's end.

Worst case scenario we merge/release this feature without GPU compatibility for WeSpeaker embeddings

Reply:
I simplified it now, e.g. removed the .cpu() part.